libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -210,6 +292,27 @@ public OAuthException(string body, Exception inner)
             }
         }
 
+        /// <summary>
+        /// Has the token expired?  If so, then we await on every attempt to get a new token
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        private bool IsTokenExpired(OAuthResponse token)
```

Comment:
a) I would make this static
b) I would make this an expression bodied method
c) I would mark it for aggressive inlining so that the JIT will make a real effort to inline it

```CSharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
 private static bool IsTokenExpired(OAuthResponse token)  =>
    DateTime.UtcNow > token.expiration_time;
```